src/mscorlib/shared/System/Text/StringBuilder.cs

```diff
@@ -1163,6 +1153,32 @@ private unsafe StringBuilder AppendJoinCore<T>(char* separator, int separatorLen
             return this;
         }
 
+        private unsafe StringBuilder AppendJoinCore<T>(char* separator, int separatorLength, T[] values)
+        {
+            if (values == null)
+                throw new ArgumentNullException(nameof(values));
+            Contract.Ensures(Contract.Result<StringBuilder>() != null);
```

Comment:
Also optional, but we can delete Contract.Ensures. We have given up on enforcing contracts, and this one cannot easily become a Debug.Assert.
https://github.com/dotnet/corefx/issues/15443